File: changer/src/main/kotlin/junny/land/xlsx/builder/extract/FieldValue.java
package junny.land.xlsx.builder.extract;

import static java.lang.Boolean.TRUE;
import static java.lang.String.valueOf;

import java.lang.reflect.Field;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;
import junny.land.xlsx.datas.FieldDatas;

public class FieldValue<T> {

    private final T objects;

    public FieldDatas extract() {
        Class<?> clazz = objects.getClass();
        List<String> datas = Arrays.stream(clazz.getDeclaredFields()).map(this::toString).collect(
            Collectors.toList());
        return new FieldDatas(datas);
    }

    public String toString(Field element) {
        try {
            element.setAccessible(TRUE);
            return valueOf(element.get(objects));
        } catch (IllegalAccessException e) {
            return "";
        }
    }

    public FieldValue(T objects) {
        this.objects = objects;
    }
}

File: changer/src/main/kotlin/junny/land/xlsx/builder/extract/FieldValue.kt
package junny.land.xlsx.builder.extract

import junny.land.xlsx.datas.FieldDatas
import java.lang.reflect.Field
import java.util.*
import java.util.stream.Collectors
import kotlin.streams.toList

class FieldValue<T>(
    private val objects: T,
    private val clazz: Class<T>
) {
    fun extract(): FieldDatas {
        val datas = clazz.declaredFields
            .map { element: Field -> toString(element) }
            .toList()
        return FieldDatas(datas)
    }

    private fun toString(element: Field): String {
        return try {
            element.isAccessible = true
            element[objects].toString()
        } catch (e: IllegalAccessException) {
            ""
        }
    }
}